### ISSUE TITLE
[Question] lambdarank and the label_gain parameters

### ISSUE THREAD

**Issue by wil70** [microsoft/LightGBM#6616]
Hello

I have a dataset with 3 classes, but classes 1 and 2 represent only 1/3=2/6 of the data and class 3 represents 2/3=4/6. I'm using objective=mutliclass.

Shall add a weight for class 1 and 2 labels so it is balance with class 3? 

Is it done via the 'lambdarank' and the 'label_gain' parameters? like label_gain=4,4,1

Thanks
--w

ref - [lambdarank and the label_gain parameters](https://lightgbm.readthedocs.io/en/stable/Advanced-Topics.html#lambdarank)

**Comment by jameslamb** [comment]
`lambdarank` is not a parameter... it's a loss function (a value to be passed to the `objective` parameter) for learning-to-rank applications.

`label_gain` is not relevant for multiclass classification... it's only used for learning-to-rank with lambdarank, as described in the docs at https://lightgbm.readthedocs.io/en/latest/Parameters.html#label_gain.

Some strategies for dealing with class imbalance for multiclass classification

* re-weight the data (as you suggested), by providing training weights
* use `multiclassova` objective + `is_unbalance=true` ([docs link](https://lightgbm.readthedocs.io/en/latest/Parameters.html#is_unbalance))
* if you have a lot of features, try using very low values of `feature_fraction` (like `<= 0.1`) ... this might help in the case where some subset of features are much more helpful for one of the minority classes
* use `LGBMClassifier` from the Python package and its `class_weight` parameter ([docs](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMClassifier.html#lightgbm.LGBMClassifier.__init__))

You can also try searching around this project's issue board, Stack Overflow, Kaggle boards, etc. ... there's much discussion about how to deal with class imbalance for multiclass problems using gradient boosted trees.

Linking some related conversations:

* https://github.com/microsoft/LightGBM/issues/3702
* https://github.com/microsoft/LightGBM/issues/5399
* https://github.com/dmlc/xgboost/issues/7688

**Comment by wil70** [comment]
Super - thanks James
I will try those over the weekend     On Thursday, August 15, 2024 at 10:02:32 PM PDT, James Lamb ***@***.***> wrote:  
 
 


lambdarank is not a parameter... it's a loss function (a value to be passed to the objective parameter) for learning-to-rank applications.

label_gain is not relevant for multiclass classification... it's only used for learning-to-rank with lambdarank, as described in the docs at https://lightgbm.readthedocs.io/en/latest/Parameters.html#label_gain.

Some strategies for dealing with class imbalance for multiclass classification
   
   - re-weight the data (as you suggested), by providing training weights
   - use multiclassova objective + is_unbalance=true (docs link)
   - if you have a lot of features, try using very low values of feature_fraction (like <= 0.1) ... this might help in the case where some subset of features are much more helpful for one of the minority classes
   - use LGBMClassifier from the Python package and its class_weight parameter (docs)

You can also try searching around this project's issue board, Stack Overflow, Kaggle boards, etc. ... there's much discussion about how to deal with class imbalance for multiclass problems using gradient boosted trees.

Linking some related conversations:
   
   - #3702
   - #5399
   - dmlc/xgboost#7688
